Lib/typing.py

```diff
@@ -2412,10 +2414,72 @@ def utf8(value: bytes) -> bytes: ...
       def utf8(value: str) -> bytes: ...
       def utf8(value):
           # implementation goes here
+
+    Overload definition can be retrieved at runtime using the
+    get_overloads() function.
     """
+    try:
+        existing = get_overloads(func)
```

Comment:
I do worry a bit about performance here, because large codebases may have many overloads. I tried to measure the time for one `overload(f)` call using timeit, and this is now 1.5 usec, while before it was 50 nsec. So ~30x slower. 

One thing I notice is that *if* the function exists (which usually happens because this is the second or further overload -- i.e. at least half of all overloads) we compute the key twice.

I think it's worth spending some time optimizing this. Not *too* much, because I realize my worry is a bit speculative. (Maybe we could ask Pradeep Kumar if he can count the number of overloads in the IG codebase?)

Reply:
Good point, I'll add a few other micro-optimizations.

Reply:
It's now about 17x slower for me:
```
In [4]: %timeit overload(f)
756 ns ± 1.88 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [6]: %timeit old_overload(f)
44.1 ns ± 0.34 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [7]: 756 / 44
Out[7]: 17.181818181818183
```